Feature/docker files

sleiss opened the pull request:
### Motivation and Context
Update docker-compose files to enable easier onboarding for new developers.

### Description
The atlassian stack is now configured in a docker-compose setup.
This allows an easier start into the Artemis development.

Also, the Gitlab+Jenkins stack is now further moved to Docker.

### Steps for Testing
Follow the instructions listed [here](https://github.com/ls1intum/Artemis/blob/feature/docker-files/doc/bamboo-bitbucket-jira/BAMBOO_BITBUCKET_JIRA_SETUP.md) and verify that the setup is clearly understandable and  works as intended.

madwau commented:
Really nice! :100: This is super useful to be able to have the same Jira/Bitbucket/Bamboo experience locally as on the test server, well done :+1:

I didn't get everything fully working yet but got pretty far with only a few pointers from @sleiss via Slack. I wrote down some things that I noticed while following the setup, see below.

When I first ran the docker compose up command, I received the following error:

```bash
$ docker-compose -f src/main/docker/atlassian.yml up -d
Creating network "artemis_artemis" with driver "bridge"
ERROR: Pool overlaps with other one on this address space
```

Turns out another (idle) network on my machine was already using the address space. A `docker network prune` solved the issue. See [here](https://github.com/maxking/docker-mailman/issues/85). Maybe you could add it to the docs in case someone else gets the same error?

Steps 1-2 (installation of Jira/Bitbucket/Bamboo with admin user creation plus `atlassian-setup.sh` shell script) worked just as described.

With step 3, I had some issues that were due to my own misconfiguration though. To me it wasn't a hundred percent clear which of the 6 different urls (3x application url + 3x display url) have to be changed. While the description in the doc is correct, I intuitively thought that probably all of those urls have to be adapted to `localhost` and the chosen ports, however, this will not work. I think it would be a good idea to add screenshots here for every single application link configuration.

Step 4 makes sense. Initially, I wasn't sure for a second if I need to create the users and groups myself but then I noticed that step 2 says that the script creates them. Maybe just to make this even more clear, you could adapt the wording in step 4 from `Add the users to the groups.` to `The script has already created users and groups but you need to manually assign the users into their respective group.`.

I think step 5 needs a little more explanation as I was only able to understand it with your help via Slack. To me it wasn't entirely clear what to do here. I think screenshots would be very helpful here and make this very easy.

Steps 6-7 were perfectly clear.

Step 8 was quite easy but there was no `Add executable` link for me initially (only once I had created an executable). There was only the more generic `Add capability` link where you had to select `Executable` as the Capability type. Maybe mention that and also screenshots would be nice to have.

Configuration of `application-artemis.yml` is well described.

Login with Jira user management enabled worked.

When I created a programming exercise, it first failed since my course was using a User Group that did not exist in Jira. It could be helpful to mention somewhere that the course groups need to match those in Jira.

When working on a programming exercise in the code editor, builds get created but no agent was picking them up. Turned out that the Jobs had Docker as a requirement which the Default Agent did not provide, thus none of the agents was able to pick up the jobs. When I configured the jobs differently to not use Docker, they were being picked up correctly by the Default agent.

I think the BASE repo builds are fine now (with 13 failing tests) but for the SOLUTION and the student repos I'm getting `Could not find test result reports in the /var/atlassian/application-data/bamboo/xml-data/build-dir/DATASTRUCTURESPROGT1730-SOLUTION-JOB1 directory.`. There's probably some more configuration needed for the location of the test result reports?

sleiss commented:
@madwau Thanks for your review! I'll implement the feedback in the next days.

I just checked and found the issue for the failing/non-starting builds: I've merged develop into this branch after testing it myself and the docker configuration for Bamboo was introduced in these merged commits.
I'll add a way to run it with this setup.

sleiss commented:
@madwau 
@F4lka and I updated the PR to implement your feedback.

I've added a check inside the BambooBuildPlanService that ensures that the builds are not run in a Docker environment if the `dev` profile is active: https://github.com/ls1intum/Artemis/pull/1290/files#diff-16299e3a1fb24179ad9cb497ebe2d0c7R128

More details can be found here: https://github.com/ls1intum/Artemis/pull/1290/files#diff-cada39c5eb101fa6ac1969d61187eafcR5

Also, in case you want to update your existing setup, it should be enough to run these commands:

```
docker-compose -f src/main/docker/atlassian.yml down
docker-compose -f src/main/docker/atlassian.yml build
docker-compose -f src/main/docker/atlassian.yml up
```

And manually adjust the Maven Capability path from `/usr` to `/artemis`.
